api/app_sph_lms/api/serializer/course_serializer.py

```diff
@@ -29,20 +30,39 @@ def get_trainee(self, obj):
 
 
 class CourseSerializer(serializers.ModelSerializer):
-    categories = serializers.SerializerMethodField()
+    category = serializers.PrimaryKeyRelatedField(
+        queryset=Category.objects.all(),
+        many=True
+    )
+
     lessons = LessonSerializer(many=True)
 
     class Meta:
         model = Course
         fields = "__all__"
 
-    def get_categories(self, obj):
-        categories = obj.coursecategory_set.all()
-        return [
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation['category'] = [
             {
-                "id": category.category.id,
-                "name": category.category.name
-            } for category in categories]
+                'id': category.id,
+                'name': category.name
+            } for category in instance.category.all()
+        ]
+        return representation
+
+    def create(self, validated_data):
+        categories_data = validated_data.pop('category')
+        lessons_data = validated_data.pop('lessons')
+
+        course = Course.objects.create(**validated_data)
+
+        for lesson_data in lessons_data:
+            lesson_data['course'] = course
+            Lesson.objects.create(**lesson_data)
+
+        course.category.set(categories_data)
+        return course
```

Comment:
Before creating the course, please check the role of the user as well, we cannot allow tranees to create a coruse, they must be trainer.

Also, please make sure to save the order of lessons as well, I see in db, its saving 0 for all lesson orders, the order must be the same as you are receiving it from FE

Reply:
I have a question regarding this Sir?, do we really need the user who can create course must be a trainer? since, in the front-end design I think we already forbid the trainee to create a course like I saw in the figma design, in the trainee section there is no create course feature  in it?

Reply:
I see, thank you sir